File: 2021/lib/solver.py
class Solver:
    def __init__(self, testInput, input):
        self.testInput = testInput
        self.input = input 
        self.problemPart = 0


    def solve(self, solution : callable, testAnswer):
        self.problemPart += 1
        print("-----Solving Part: {}".format(self.problemPart))
        testOutput = solution(self.testInput)
        print("test pass: \t {} \n \t expected: {} \n \t got: {}".format(testAnswer == testOutput, testAnswer, testOutput))
        output  = solution(self.input)
        print("answer:\n \t {}".format(output))

File: 2022/04/Solver.cs
public class Solver {
    private string day;
    private string expectedTestOutput;
    private string testFileName;
    private string inputFileName;
    public Solver(string day, string expectedTestOutput, string testFileName = "test.txt", string inputFileName = "input.txt") {
        this.day = day;
        this.expectedTestOutput = expectedTestOutput;
        this.testFileName = testFileName;
        this.inputFileName = inputFileName;
    }
    public void Run() {
        Console.WriteLine("Solving day {0}", this.day);
        Console.WriteLine("Test Output: {0}, expected: {1}", this.expectedTestOutput, Solve(this.testFileName));
        Console.WriteLine("\nSovled output: {0}", Solve(this.inputFileName));
    }
    public int Solve(string fileName) {
        return 0;
    }
}